cookbooks/env_vars/recipes/cloud.rb

```diff
@@ -8,21 +8,24 @@
   perform_restart = node['env_vars']['perform_restart']
 
   node['dna']['engineyard']['environment']['apps'].each do |app_data|
-    template "/data/#{app_data['name']}/shared/config/env.cloud" do
+    app_name = app_data['name']
+
+    template "/data/#{app_name}/shared/config/env.cloud" do
       source "env.cloud.erb"
       owner ssh_username
       group ssh_username
       mode 0744
       variables(:environment_variables => fetch_environment_variables(app_data))
       helpers(EnvVars::Helper)
 
-      notifies :run, "execute[restart_#{app_data['name']}]", :delayed
+      notifies :run, "execute[restart_#{app_name}]", :delayed
     end
 
-    execute "restart_#{app_data['name']}" do
-      command "if [ -d /data/#{app_data['name']}/current ]; then /engineyard/bin/app_#{app_data['name']} restart; fi"
+    execute "restart_#{app_name}" do
+      command "/engineyard/bin/app_#{app_name} restart"
       user ssh_username
       action :nothing
+      only_if { perform_restart && ::File.exist?("/data/#{app_name}/current") && ::File.exist?("/engineyard/bin/app_#{app_name}")}
```

Comment:
@radamanthus, I think it won't work with this `perform_restart` flag. And it doesn't make sense to check it here. As I already mentioned this command will run only if config file changed and I have no idea how to pass `perform_restart = true` here. May be I missing something, but I don't how to pass it when you click to 'Apply' button on the environment page.

Reply:
@rzaharenkov  perform_restart is set here (line 8):

```
perform_restart = node['env_vars']['perform_restart']
```

node['env_vars']['perform_restart'] in turn is set here: https://github.com/engineyard/ey-cookbooks-stable-v5/blob/next-release/custom-cookbooks/env_vars/cookbooks/custom-env_vars/attributes/default.rb#L11

We want the additional perform_restart check in place because we want the customer to control if an application restart should be done during the chef run. That's my thinking anyway, and I'm open to be convinced otherwise. (e.g. Customers might get confused - "I changed the env var values, why am I not seeing them reflect in my application?")

Reply:
As of now I prefer "avoid unintended downtime" over "confusion on why env vars do not get applied".